src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -2754,14 +2795,17 @@ Status ColorControlServer::moveToColorTempCommand(EndpointId endpoint,
  * @param commandData Struct containing the parameters of the command
  * @return Status::Success when successful,
  *         Status::InvalidCommand when stepSize is 0 or an unknown stepMode is provided
- *         Status::UnsupportedEndpoint when the provided endpoint doesn't correspond with a color temp transition state.
+ *         Status::UnsupportedEndpoint when the provided endpoint doesn't correspond with a color temp transition state,
+ *         Status::ConstraintError when a command parameters is outside its defined value range.
  */
 Status ColorControlServer::stepColorTempCommand(EndpointId endpoint,
                                                 const Commands::StepColorTemperature::DecodableType & commandData)
 {
     // Confirm validity of the step mode and step size received
     VerifyOrReturnValue(commandData.stepMode != HueStepMode::kUnknownEnumValue, Status::InvalidCommand);
     VerifyOrReturnValue(commandData.stepSize != 0, Status::InvalidCommand);
+    // Command Parameters constraint checks:
+    VerifyOrReturnValue(commandData.transitionTime <= kMaxtransitionTime, Status::ConstraintError);
```

Comment:
Needs to come before the stepSize check.

And needs constraint checks on `colorTemperatureMinimumMireds` and `colorTemperatureMaximumMireds` as well, no?

Reply:
Moving the constraint check first. Same thing as `moveToColorTempCommand` happens here the current implementation, the `ColorTempPhysicalMinMireds` and ColorTempPhysicalMaxMireds gat the value used but I can still add the constraint check